### PR TITLE
Roughly fulfill the request for a demon den villain from wiki

### PR DESCRIPTION
**Pull request by keepermanj**
The demon den was requested here on the wiki:
http://keeperrl.com/wiki/index.php?title=Requests_for_new_villains#Demon_Den_.28Demons.29

I roughly followed the request but I changed some things as I thought best. The demon den only appears on adventurer maps at the moment. The demons are quite tough, so there is only one level, which is below ground.

What do you think of this Michal?

[The commits on 2nd and 3rd of August are redundant and seem to want to keep coming back without actually doing anything. Sorry about this.]

### CODE REVIEW

**Comment by miki151** [comment]
At this point there is nothing special about this villain, I don't feel it adds anything compared to existing villains, like knights. Are there any special attacks or custom AI that you think they could have?

Also you didn't add any attack triggers, was it on purpose?

**Comment by keepermanj** [comment]
I have thought about it and I want to make the demons attack if they are in proximity and you have not built enough demon shrines to keep them happy. This will need a new type of attack trigger for furniture that is required for a truce.

The way it is at the moment, the demons are for adventurers to kill, which is why they don't have attack triggers. The Knights are only for keepers to kill The demons also have magic resistance and the circular blast spell, so there is nothing else exactly like them.

Please give me your thoughts, so I can make it so :)

File: village_behaviour.cpp [hunk 1]
```diff
@@ -101,6 +101,7 @@ static double stolenItemsFun(int numStolen) {
 static double getRoomProb(FurnitureType id) {
   switch (id) {
     case FurnitureType::THRONE: return 0.001;
+	case FurnitureType::DEMON_SHRINE: return 0.001;
```
Comment (miki151) [inline village_behaviour.cpp]: Please fix formatting.

File: village_behaviour.cpp [hunk 2]
```diff
@@ -140,8 +141,14 @@ double VillageBehaviour::getTriggerValue(const Trigger& trigger, const VillageCo
     switch (trigger.getId()) {
       case AttackTriggerId::TIMER: 
         return collective->getGlobalTime() >= trigger.get<int>() ? timerProb : 0;
-      case AttackTriggerId::ROOM_BUILT: 
-        return collective->getConstructions().getBuiltCount(trigger.get<FurnitureType>()) *
+      case AttackTriggerId::ROOM_BUILT:
+		if (trigger.get<FurnitureType>() ==FurnitureType::DEMON_SHRINE)
```
Comment (miki151) [inline village_behaviour.cpp]: Please fix formatting. Use double space for indentation, no tabs. Thank you :)

**Pull request merged**